src/components/Tree/components/EditableNodeItem.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ChangeEvent, KeyboardEvent, useState } from "react";
+
+export interface EditableNodeItem {
+    name: string;
+    onEditableSave: (value: string) => void;
+}
+
+export const EditableNodeItem = ({ name, onEditableSave }: EditableNodeItem) => {
+    const [inputValue, setInputValue] = useState(name);
+    const [showInput, setShowInput] = useState<boolean>(false);
+
+    const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === "Enter") {
+            setShowInput(false);
+
+            onEditableSave && onEditableSave((event.target as HTMLInputElement).value);
```

Comment:
The interface defined the property not as optional, i think you can remove the check with the &&